Add documentation of user cases (#155)
megaease/easegress: doc/cookbook/api_aggregator.md

`pipeline-demo` will return `{"mega":"ease"}` in HTTP response body.
`pipeline-demo1` will return `{"hello":"world"}`.
`pipeline-demo2` will return `{"hello":"new world"}`.

``` bash
echo '
name: pipeline-api
kind: HTTPPipeline
flow:
  - filter: agg
filters:
  - name: agg
    kind: APIAggregator
    pipelines:
      - name: pipeline-demo
      - name: pipeline-demo1
      - name: pipeline-demo2 '  | egctl object create
```

2. Creating an HTTPServer for forwarding the traffic to this pipeline.

``` bash
echo '
kind: HTTPServer
name: server-demo
port: 10080
keepAlive: true
https: false
rules:
  - paths:
    - pathPrefix: /api
      backend: pipeline-api | egctl object create

```

3. Request the aggregator pipeline

``` bash

$ curl  -X GET  http://127.0.0.1:10080/api -v
[{"hello":"world"},{"hello":"new world"},{"mega":"ease"}]

```

### Scenario 2: Merge response body

* As in #Scenario 1,  `pipeline-demo1` and `pipeline-demo2`'s responses share the same JSON key, we want to merge their response body by the JSON key together. If the keys have conflicted, we will use the last value.

> 1. Update the pipeline with aggregator

Reviewer: ```suggestion
1. Update the pipeline with the aggregator
```

``` bash
echo '
name: pipeline-api
kind: HTTPPipeline
flow:
  - filter: agg
filters:
  - name: agg
    kind: APIAggregator
    mergeResponse: true   // turn on this switch
    pipelines:
      - name: pipeline-demo
      - name: pipeline-demo1
      - name: pipeline-demo2 '  | egctl object update
```

2. Request the aggregator pipeline

``` bash

$ curl  -X GET  http://127.0.0.1:10080/api -v
{"hello":"new world","mega":"ease"}

```

### Scenario 3: Allow partial succeed

> * As In #Scenario 1, if we want the backend service meets some problem and can't provide service, this aggregator will
> failed as blew:

Reviewer: ```suggestion
* As In #Scenario 1, if we want the backend service to meet some problem and can't provide service, this aggregator will be failed as blew:
```

``` bash
$ curl http://localhost:10080/api -v -X PUT
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 10080 (#0)
> PUT /api HTTP/1.1
> Host: localhost:10080
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 503 Service Unavailable
< X-Eg-Aggregator: failed-in-pipeline-demo
< Date: Wed, 28 Jul 2021 09:11:18 GMT
< Content-Length: 0
<
* Connection #0 to host localhost left intact
* Closing connection 0
```

> * We can see an `X-Eg-Aggregator: failed-in-pipeline-demo` header in response, that's the first met failure pipeline of this aggregator.

Reviewer: ```suggestion
* We can see an `X-Eg-Aggregator: failed-in-pipeline-demo` header in the response, that's the first met failure pipeline of this aggregator.
```
